### PR TITLE
setup file added

### PR DESCRIPTION
**Pull request by allaffa**
Add `setup.py`. Add installation files to `.gitignore`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from distutils.core import setup
+
+# Note: setup() has access to cmd arguments of the setup.py script via sys.argv
+
+setup(
+    name="HydraGNN",
+    version="1.0",
```
Comment (streeve) [inline setup.py:7]: I would start with 0.1-dev or 0.1

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    package_dir={"hydragnn": "hydragnn"},
+    packages=["hydragnn"],
+    author="Massimiliano Lupo Pasini, Samuel Temple Reeve, Pei Zhang, Jong Youl Choi",
```
Comment (streeve) [inline setup.py:10]: Marko?

**Review by streeve — CHANGES_REQUESTED** [review]

**Comment by streeve** [comment]
What you started with is probably sufficient, but we may as well add other easy categories (e.g. license): https://pythonhosted.org/an_example_pypi_project/setuptools.html 

Do you want to replace `requirements.txt` or keep both?

File: setup.py (hunk 2 above)
Comment (allaffa) [reply setup.py:10]: He is nor an author of the repository

**Comment by allaffa** [comment]
> What you started with is probably sufficient, but we may as well add other easy categories (e.g. license): https://pythonhosted.org/an_example_pypi_project/setuptools.html
> 
> Do you want to replace `requirements.txt` or keep both?

@streeve I would like to keep both

File: setup.py (hunk 1 above)
Comment (streeve) [reply setup.py:7]: After discussing we can leave as is and plan to release after current PRs are mereged

File: setup.py (hunk 2 above)
Comment (streeve) [reply setup.py:14]: Not actively, but his contributions are still in the repo

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# Note: setup() has access to cmd arguments of the setup.py script via sys.argv
+
+setup(
+    name="HydraGNN",
+    version="1.0",
+    package_dir={"hydragnn": "hydragnn"},
+    packages=["hydragnn"],
+    description="Distributed PyTorch implementation of multi-headed graph convolutional neural networks",
+    license="BSD-3",
+    long_description=read("README"),
```
Comment (streeve) [inline setup.py:12]: Need to add the `read` helper function

**Review by streeve — CHANGES_REQUESTED** [review]

File: setup.py (hunk 2 above)
Comment (allaffa) [reply setup.py:14]: The list of people added here is going to reflect the name of people listed as authors in the technical transfer.

**Review by pzhanggit — APPROVED** [review]

**Review by streeve — APPROVED** [review]

**Pull request merged**